sktime/transformations/series/hpfilter.py

```diff
@@ -0,0 +1,123 @@
+    >>> cycle, trend = sm.tsa.filters.hpfilter(dta.realgdp, 1600)
+    """
+
+    _tags = {
+        "scitype:transform-input": "Series",
+        # what is the scitype of X: Series, or Panel
+        "scitype:transform-output": "Series",
+        # what scitype is returned: Primitives, Series, Panel
+        "scitype:instancewise": True,  # is this an instance-wise transform?
+        "univariate-only": True,  # can the transformer handle multivariate X?
+        "X_inner_mtype": "np.ndarray",  # which mtypes do _fit/_predict support for X?
+        # this can be a Panel mtype even if transform-input is Series, vectorized
+        "y_inner_mtype": "None",  # which mtypes do _fit/_predict support for y?
+        "requires_y": False,  # does y need to be passed in fit?
+        "enforce_index_type": [
+            pd.RangeIndex
+        ],  # index type that needs to be enforced in X/y
+        "fit_is_empty": True,  # is fit empty and can be skipped? Yes = True
+        "transform-returns-same-time-index": False,
+        # does transform return have the same time index as input X
+        "capability:unequal_length": True,
+        # can the transformer handle unequal length time series (if passed Panel)?
+        "handles-missing-data": False,  # can estimator handle missing data?
+        "remember_data": False,  # whether all data seen is remembered as self._X
+        "python_dependencies": "statsmodels",
+    }
+
+    def __init__(
+        self,
+        lamb=1600,
+    ):
+        self.lamb = lamb
+        super(HPFilter, self).__init__()
+
+    def _transform(self, X, y=None):
+        """Transform X and return a transformed version.
+        private _transform containing core logic, called from transform
+        Parameters
+        ----------
+        X : array_like, A 1d array
+
+        Returns
+        -------
+        transformed cyclical version of X
+        """
+        from statsmodels.tsa.filters.hp_filter import hpfilter
+
+        kwargs = {"lamb": self.lamb}
+        return hpfilter(X, **kwargs)[0]
+
+    @classmethod
+    def get_test_params(cls, parameter_set="default"):
+        """Return testing parameter settings for the estimator.
+        Parameters
+        ----------
+        parameter_set : str, default="default"
+            Name of the set of test parameters to return, for use in tests. If no
+            special parameters are defined for a value, will return `"default"` set.
+            There are currently no reserved values for transformers.
+        Returns
+        -------
+        params : dict or list of dict, default = {}
+            Parameters to create testing instances of the class
+            Each dict are parameters to construct an "interesting" test instance, i.e.,
+            `MyClass(**params)` or `MyClass(**params[i])` creates a valid test instance.
+            `create_test_instance` uses the first (or only) dictionary in `params`
+        """
+        params = {"lamb": 1600}
```

Comment:
there should always be at least two parameter sets.

I would go with an empty dict (the default) and this one.